openbsd/OpenBSDMachine.c

```diff
@@ -129,7 +130,7 @@ void Machine_delete(Machine* super) {
 }
 
 static void OpenBSDMachine_scanMemoryInfo(OpenBSDMachine* this) {
-   Machine* host = &this->super;
+   Machine* super = (Machine*)&this->super;
```

Comment:
There should be no cast necessary. Please provide build message.

Reply:
iirc it was saying there was a cast needed, which is odd. I didn't think that was necessary in C.

I'll revert the casts and show you the errors I'm getting.

Reply:
Ah, i see now; they were warnings, not errors. In that case I'll revert that.

Reply:
Can you post the warnings you are getting? Just to check if there's anything funny going on.

Reply:
It's the same warning that can be found in other sources I did not modify, such as this one:
```
OpenBSDProcessList.c:41:11: warning: incompatible pointer types returning 'OpenBSDProcessList *' (aka 'struct OpenBSDProcessList_ *') from a function with result type 'ProcessList *' (aka 'struct ProcessList_ *') [-Wincompatible-pointer-types]
```

There seems to be 3 other instances of this warning:
```
OpenBSDMachine.c:117:11: warning: incompatible pointer types returning 'OpenBSDMachine *' (aka 'struct OpenBSDMachine_ *') from a function with result type 'Machine *' (aka 'struct Machine_ *') [-Wincompatible-pointer-types]
```

```
OpenBSDMachine.c:121:20: warning: incompatible pointer types initializing 'OpenBSDMachine *' (aka 'struct OpenBSDMachine_ *') with an expression of type 'Machine *' (aka 'struct Machine_ *') [-Wincompatible-pointer-types]
```

```
OpenBSDMachine.c:265:20: warning: incompatible pointer types initializing 'OpenBSDMachine *' (aka 'struct OpenBSDMachine_ *') with an expression of type 'Machine *' (aka 'struct Machine_ *') [-Wincompatible-pointer-types]
```